### DISCUSSION TITLE
Where to find extensive log and get notified ASAP for a bad login?

### DISCUSSION THREAD

**Discussion by adnan29979** [Ylianst/MeshCentral#5810] [Q&A] [answered]
I am searching for a comprehensive and detailed log in MeshCentral, specifically related to login attempts, both valid and invalid. Also a way to immediately notify me via email about any unsuccessful login attempts. Currently, it seems we cannot view the username if it's not registered. I wish to view almost everything related to login activities. Is there a way to do this, or does MeshCentral not yet offer such a feature?

And can someone kindly explain how to generate the log in the screenshot to me? (I sourced it from a Reddit post).

**Comment by si458** [comment, marked as answer]
You can set `authLog` to a filepath+filename under `settings` of ur config.json
This can then be phased with the likes of fail2ban
![Screenshot_20240216_232554_Firefox](https://github.com/Ylianst/MeshCentral/assets/765314/2eb6b99a-e17b-4c37-82ba-8ed689b4a2f7)

**Reply by si458** [reply]
I'm not aware of other ways at the moment of being notified
If someone tried to login with a valid user but wrong password, then that user will get notified in there my events and web notification in the ui
But again no email notifications I'm afraid
You would have to open a githib enhancement issue and we would look into adding this someway, somehow

**Reply by adnan29979** [reply]
Yes. The authLog would help a lot. I was unaware of this. Thanks a lot for showing this. I might set up some separate thing on the same machine to notify me of invalid login attempts from this log file's data. Thanks a lot again. ❤
Another question. Is there any limit as to how many day's worth of log will be saved in this authLog file?

And apologies, I forgot to add the Reddit screenshot I was talking about. 
![image](https://github.com/Ylianst/MeshCentral/assets/148310766/cf7b13ea-3c5f-415c-ba0d-4ef30811a320)

**Reply by si458** [reply]
In theory, no limit, it just keeps adding to the text file
And in theory the format is `date, username -> info, useragent` will check later for you

**Reply by si458** [reply]
the format is
```
const d = new Date(), month = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'][d.getMonth()];
msg = month + ' ' + d.getDate() + ' ' + obj.common.zeroPad(d.getHours(), 2) + ':' + obj.common.zeroPad(d.getMinutes(), 2) + ':' + d.getSeconds() + ' meshcentral ' + server + '[' + process.pid + ']: ' + msg + ((obj.platform == 'win32') ? '\r\n' : '\n');
```
so 
```
Feb 15 02:25:0 meshcentral https[407]: Failed password for undefined from 185.91.xxx.xxx port 35630
Feb 15 12:12:16 meshcentral https[407]: Accepted password for admin from 79.77.xxx.xxx port 63566
```
